From eb59091da371d02aade0b5f325e1fee1f8a3bb99 Mon Sep 17 00:00:00 2001
From: Stephan Gerhold <stephan.gerhold@linaro.org>
Date: Tue, 22 Jul 2025 09:24:36 +0200
Subject: [PATCH 21/28] arm64: dts: qcom: x1-el2: Add iris video-firmware node

Add the video-firmware node with the IOMMU used for loading the Iris
firmware to make video acceleration work when running in EL2.

In EL2 we can load the generic firmware, so there is no need to keep Iris
disabled by default. The generic firmware is used to validate the Iris
driver and will often be more reliable, so override the firmware-name if
needed to use that even if a device-specific firmware-name is defined.

Signed-off-by: Stephan Gerhold <stephan.gerhold@linaro.org>
---
 arch/arm64/boot/dts/qcom/x1-el2.dtso | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/x1-el2.dtso b/arch/arm64/boot/dts/qcom/x1-el2.dtso
index d5e6f8fbe..9c31031d1 100644
--- a/arch/arm64/boot/dts/qcom/x1-el2.dtso
+++ b/arch/arm64/boot/dts/qcom/x1-el2.dtso
@@ -17,8 +17,13 @@ &gpu_zap_shader {
 };
 
 &iris {
-	/* TODO: Add video-firmware iommus to start IRIS from EL2 */
-	status = "disabled";
+	/* Use generic firmware in EL2 and enable it by default */
+	status = "okay";
+	firmware-name = "qcom/vpu/vpu30_p4.mbn";
+
+	video-firmware {
+		iommus = <&apps_smmu 0x1942 0x0>;
+	};
 };
 
 /*
